redis/src/cluster_async/mod.rs

```diff
@@ -1108,15 +1137,38 @@ where
         }
     }
 
+    /// This function takes a node's address, examines if its host has encountered a DNS change, where the node's endpoint now leads to a different IP address.
+    /// If no socket addresses are discovered for the node's host address, or if it's a non-DNS address, it returns false.
+    /// In case the node's host address resolves to socket addresses and none of them match the current connection's IP, a DNS change is detected, resulting in a true return.
+    async fn is_dns_changed(addr: &str, curr_ip: &IpAddr) -> bool {
+        let (host, port) = match get_host_and_port_from_addr(addr) {
+            Some((host, port)) => (host, port),
+            None => return false,
+        };
+        let mut updated_addresses = match get_socket_addrs(host, port).await {
+            Ok(socket_addrs) => socket_addrs,
+            Err(_) => return false,
+        };
+
+        !updated_addresses.any(|socket_addr| socket_addr.ip() == *curr_ip)
+    }
+
     async fn get_or_create_conn(
         addr: &str,
-        conn_option: Option<ConnectionFuture<C>>,
+        conn_option: Option<ClusterNode<C>>,
         params: &ClusterParams,
-    ) -> RedisResult<C> {
-        if let Some(conn) = conn_option {
-            let mut conn = conn.await;
+    ) -> RedisResult<(C, Option<IpAddr>)> {
+        if let Some(node) = conn_option {
+            let mut conn = node.connection.await;
+            if let Some(ref ip) = node.ip {
+                // Check for a DNS change
```

Comment:
minor: IMO the doc here just repeats the name of the function and isn't necessary.